src/openjd/model/_internal/_validator_functions.py

```diff
@@ -0,0 +1,117 @@
+
+from .._format_strings import FormatString
+
+
+def validate_int_fmtstring_field(
+    value: Union[int, float, Decimal, str, FormatString], ge: Optional[int] = None
+) -> Union[int, float, Decimal, str, FormatString]:
+    """Validates a field that is allowed to be either an integer, a string containing an integer,
+    or a string containing expressions that resolve to an integer."""
+    # Validate the type
+    if isinstance(value, str):
+        if not isinstance(value, FormatString):
+            value = FormatString(value)
+        # If the string value has no expressions, we can validate the value now.
+        if len(value.expressions) == 0:
+            try:
+                int_value = int(value)
+            except ValueError:
+                raise ValueError("String literal must contain an integer.")
+        else:
+            # In this case, we cannot validate now. We need to validate this later when the template
+            # is instantiated into a job and the expressions have been resolved.
+            return value
+    elif isinstance(value, (int, float, Decimal)) and not isinstance(value, bool):
+        int_value = int(value)
+        if int_value != value:
+            raise ValueError("Value must be an integer or a string containing an integer.")
+    else:
+        raise ValueError("Value must be an integer or a string containing an integer.")
+
+    # Validate the value constraints
+    if ge is not None and not (int_value >= ge):
+        raise PydanticKnownError("greater_than_equal", {"ge": ge})
+
+    return int_value
+
+
+def validate_float_fmtstring_field(
+    value: Union[int, float, Decimal, str, FormatString], ge: Optional[Decimal] = None
+) -> Union[int, float, Decimal, str, FormatString]:
+    """Validates a field that is allowed to be either an float, a string containing an float,
+    or a string containing expressions that resolve to a float."""
+    # Validate the type
+    if isinstance(value, str):
+        if not isinstance(value, FormatString):
+            value = FormatString(value)
+        # If the string value has no expressions, we can validate the value now.
+        if len(value.expressions) == 0:
+            try:
+                float_value = Decimal(str(value))
+            except InvalidOperation:
+                raise ValueError("Value must be floating point.")
+        else:
+            # In this case, we cannot validate now. We need to validate this later when the template
+            # is instantiated into a job and the expressions have been resolved.
+            return value
+    elif isinstance(value, (int, float, Decimal)) and not isinstance(value, bool):
+        try:
+            float_value = Decimal(str(value))
+        except InvalidOperation:
+            raise ValueError("Value must be floating point.")
+    else:
+        raise ValueError("Value must be a float or a string containing a float.")
```

Comment:
Is there a reason we aren't using a consistent error message about what the value must be?